script/deploy/healthCheck.ts

```diff
@@ -239,278 +239,356 @@ const main = defineCommand({
+          // Normal flow - check against both config and on-chain data
+          for (const cfgAddress of expectedWhitelistedAddresses) {
+            if (!cfgAddress) {
+              logError(`Encountered undefined whitelisted address.`)
+              continue
+            }
+
+            try {
+              const normalized = getAddress(cfgAddress)
+              if (!onChainWhitelisted.includes(normalized)) {
+                logError(
+                  `Whitelisted address ${normalized} not whitelisted in Diamond`
+                )
+                numMissing++
+              }
+            } catch (err) {
+              logError(`Invalid whitelisted address in config: ${cfgAddress}`)
+            }
+          }
 
-        try {
-          const normalized = getAddress(dex)
-          if (!approvedDexs.includes(normalized)) {
-            logError(`DEX ${normalized} not approved in Diamond`)
-            numMissing++
+          for (const name of autoWhitelistPeripheryContracts) {
+            // get address from deploy log
+            const addr = deployedContracts[name]
+            if (!addr) {
+              logError(`Periphery contract ${name} not deployed`)
+              numMissing++
+              continue
+            }
+
+            // check if address is whitelisted
+            const normalized = getAddress(addr)
+            if (!onChainWhitelisted.includes(normalized)) {
+              logError(`Periphery contract ${name} not whitelisted`)
+              numMissing++
+            } else consola.success(`Periphery contract ${name} whitelisted`)
           }
-        } catch (err) {
-          logError(`Invalid DEX address in main check: ${dex}`)
         }
-      }
 
-      // Ensure that periphery contracts which are used like DEXs are whitelisted
-      for (const name of autoWhitelistPeripheryContracts) {
-        // get address from deploy log
-        const addr = deployedContracts[name]
-        if (!addr) {
-          logError(`Periphery contract ${name} not deployed`)
-          numMissing++
-          continue
-        }
+        consola.info(
+          `Found ${numMissing} missing whitelisted address${
+            numMissing === 1 ? '' : 's'
+          }`
+        )
 
-        // check if address is whitelisted
-        const normalized = getAddress(addr)
-        if (!approvedDexs.includes(normalized)) {
-          logError(`Periphery contract ${name} not approved as a DEX`)
-          numMissing++
-        } else consola.success(`Periphery contract ${name} approved as a DEX`)
-      }
+        //          ╭─────────────────────────────────────────────────────────╮
+        //          │                   Check approved sigs                   │
+        //          ╰─────────────────────────────────────────────────────────╯
 
-      consola.info(
-        `Found ${numMissing} missing dex${numMissing === 1 ? '' : 's'}`
-      )
+        consola.box('Checking DEX signatures approved in diamond...')
```

Comment:
didnt we also want to rename sigs > selectors?

Reply:
good point!